handbook/Best practices/TypeScript/Project-References.md

```diff
@@ -0,0 +1,166 @@
+# Project References
+
+## Table of contents
+
+1. [Why use Project References?](#why-use-project-referencesgi)
+1. [Migrating a codebase](#migrating-a-codebase)
+1. [Tips](#tips)
+1. [Caveats](#caveats)
+
+
+## Why use Project References?
+
+Project References provides built-in scalability for TypeScript. It does this by providing developers tools to separate their code into smaller blocks. This enforces logical guidelines across the codebase which also results in healthier code over time. 
```

Comment:
If we have a "what are project references" section we can cut out the explanation point and just have a little paragraph for the two main benefits (performance and code structure)

```markdown
### Performance
Project references enable [incremental compiles](something that backs this up) and drastically [reduce type-check times](something that backs this uo) in the editor...

### Code structure
Project references create a strong enforcement of component / sub-project boundaries...
```